### PR TITLE
Use site configs in Demo

### PR DESCRIPTION
**Pull request by fraxachun**
## Description

Adapt the latest changes in comet-start to the demo.

Now demo-admin, demo-api and demo-site use the site-configs. However, the secondary site-config still leads to demo-pages.

Furthermore, the demo-site now uses domain-rewrites, which is accomplished by the middleware and requires a change in the folder structure. Nevertheless, the changes are very similar to that what we use in the starter.

However, some things are still not like in starter
- public/assets does not exist
- dam-rewrite
- layout with header ist different (if we harmonize it, in a separate PR which is not from me)

Supersedes https://github.com/vivid-planet/comet/pull/2687

-   [X] I have verified if my change requires a changeset -> No changeset because no change in Comet itself

## Further information

https://vivid-planet.atlassian.net/browse/COM-1193

### CODE REVIEW

File: demo/api/src/site-configs.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    public: {
+        scope: {
+            domain: string;
+            languages: string[];
+        };
+    };
+}
+
+export type PrivateSiteConfig = ExtractPrivateSiteConfig<SiteConfig>;
+export type PublicSiteConfig = ExtractPublicSiteConfig<SiteConfig>;
```
Comment (nsams) [inline demo/api/src/site-configs.d.ts:18]: this file should be a symlink to demo/site-configs/site-configs.d.ts, not a copy

File: demo/site-configs/site-configs.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        domains: {
+            main: "localhost:3000",
+        },
+        public: {
+            scope: {
+                domain: "main",
+                languages: ["en", "de"],
+            },
+        },
+    },
```
Comment (nsams) [inline demo/site-configs/site-configs.ts:16]: should we also split that into individual files (per domain) to be as similar as possible to starter?

**Comment by nsams** [comment]
`npm run create-site-configs-env` should be added to install.sh i assume

File: demo/api/src/site-configs.d.ts (hunk 1 above)
Comment (nsams) [reply demo/api/src/site-configs.d.ts:18]: no, it is not. The copy-schema-files.js you linked is executed in pipelines only.

The file should not be duplicated in the git repository, with the need to apply any changes 3 times.

File: demo/site-configs/site-configs.ts (hunk 2 above)
Comment (nsams) [reply demo/site-configs/site-configs.ts:16]: Imho we should use in demo the same approach as we have in starter.

And you can start a (independent of this PR) discussion (or create a PR) for changing the approach at any time.

**Review by nsams — DISMISSED** [review]

File: demo/site/src/middleware.ts [hunk 3]
```diff
@@ -29,7 +36,25 @@ export async function middleware(request: NextRequest) {
...
 
-    return NextResponse.next();
+    const predefinedPageRedirect = await getPredefinedPageRedirect(siteConfig.scope.domain, pathname);
+    if (predefinedPageRedirect) {
+        return NextResponse.redirect(new URL(predefinedPageRedirect, request.url), 307);
+    }
+
+    const predefinedPageRewrite = await getPredefinedPageRewrite(siteConfig.scope.domain, pathname);
+    if (predefinedPageRewrite) {
+        return NextResponse.rewrite(new URL(predefinedPageRewrite, request.url));
```
Comment (johnnyomair) [inline demo/site/src/middleware.ts:46]:
```suggestion
        return NextResponse.rewrite(new URL(`/${siteConfig.scope.domain}${predefinedPageRewrite}`, request.url));
```

This needs the domain prefix to work correctly.

**Review by johnnyomair — COMMENTED** [review]
Please fix conflicts.

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Review by thomasdax98 — APPROVED** [review]

**Pull request merged**